### ISSUE TITLE
$localize usage in component is reset on every extraction

### ISSUE THREAD

**Issue by spock123** [martinroob/ngx-i18nsupport#176]
UPDATE UPDATE

I noticed that if the translation state  is set to "`translated`", it works as designed.
If the state = `final` it will be overwritten on each extraction.
-------------------------

Hi, I'm using $localize to generate translations inside component classes.

Example:

`public order =  $localize`:@@order.order-step-label:`;`

However, after running extraction, while the field is found and inserted in the messages file,  both source and target for the translations are empty:

` <source/><target state="final"/>`

Further, if we translate it and update the translation file, on the next extraction run, the translations are reset to the initial empty state again.

Any ways we can prevent this? It makes it impossible to use the tool as all our translations are removed each time we run the script.

Configuration:

```
{
  "xliffmergeOptions": {
    "srcDir": "src/locale",
    "genDir": "src/locale",
    "i18nFile": "messages.xlf",
    "i18nBaseFile": "messages",
    "i18nFormat": "xlf",
    "encoding": "UTF-8",
    "defaultLanguage": "en",
    "languages": ["en", "da"],
    "removeUnusedIds": true,
    "supportNgxTranslate": false,
    "ngxTranslateExtractionPattern": "@@",
    "useSourceAsTarget": true,
    "targetPraefix": "",
    "targetSuffix": "",
    "beautifyOutput": false,
    "preserveOrder": true,
    "allowIdChange": false,
    "autotranslate": false,
    "apikey": "",
    "apikeyfile": "",
    "verbose": false,
    "quiet": false
  }
}
```

**Comment by julkue** [comment]
The problem is, it's only compatible with Angular v8 currently, where `$localize` didn't even exist. See https://github.com/martinroob/ngx-i18nsupport/issues/167

**Comment by ahmadalfy** [comment]
@julmot it's compatible with 10 as well

**Comment by julkue** [comment]
@ahmadalfy Yes, the base functionality that was already present with v8 kept working up and including to v10 (not v11). But it's not officially compatible, thus a feature like `$localize` that wasn't there in v8 isn't officially supported, see: https://github.com/martinroob/ngx-i18nsupport/blob/master/package.json#L30

**Comment by ahmadalfy** [comment]
@julmot you're right, I found that I am not using it. I ended up removing it from my project and used [xliffmerge](https://github.com/martinroob/ngx-i18nsupport/tree/master/projects/xliffmerge) from the same author.
